src/nplinker/metabolomics/gnps/gnps_downloader.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+import httpx
+
+from .gnps_format import GNPSFormat
+from .gnps_format import gnps_format_from_task_id
+
+
+class GNPSDownloader:
+    GNPS_DATA_DOWNLOAD_URL = 'https://gnps.ucsd.edu/ProteoSAFe/DownloadResult?task={}&view=download_clustered_spectra'
+    GNPS_DATA_DOWNLOAD_URL_FBMN = 'https://gnps.ucsd.edu/ProteoSAFe/DownloadResult?task={}&view=download_cytoscape_data'
+
+    def __init__(self, task_id: str, download_root: Path):
+        """Init the class with a gnps_task_id and a output path on where to save the data.
+
+        Args:
+            task_id(str): GNPS task id, identifying the data to be downloaded.
+            download_root(Path): Path where to store the downloaded archive.
+
+        Examples:
+            >>> GNPSDownloader("c22f44b14a3d450eb836d607cb9521bb", "~/downloads")
+            """
+        self._task_id = task_id
+        self._download_root = download_root
+
+
+    def download(self):
+        """Execute the downloading process. """
+        with open(self.target(), 'wb') as f:
+            with httpx.stream('POST', self.url()) as r:
+                for data in r.iter_bytes():
+                    f.write(data)
+   
+    def target(self) -> Path:
+        return self._download_root.joinpath(self._task_id + ".zip")
+    
+    def task_id(self) -> str:
```

Comment:
It looks like a convention in python that how we use class methods: `obj.verb_noun()` and `obj.noun`.
So I prefer renaming `task_id` to `get_task_id`, or changing `task_id` to a property. 
Other methods should also be updated in a similar way.

Reply:
Ok, I'll do that.